source/game/system/map/MapdataCannonPoint.cc

```diff
@@ -0,0 +1,47 @@
+#include "MapdataCannonPoint.hh"
+
+#include "game/system/CourseMap.hh"
+
+namespace System {
+
+const EGG::Vector3f &MapdataCannonPoint::pos() const {
+    return m_pos;
+}
+
+const EGG::Vector3f &MapdataCannonPoint::rot() const {
+    return m_rot;
+}
+
+u16 MapdataCannonPoint::id() const {
+    return m_id;
+}
+
+const MapdataCannonPoint::CannonParameter &MapdataCannonPoint::parameters() const {
+    assert(m_paramterIdx <= 2);
+    const s16 idx = m_paramterIdx < 1 ? 0 : m_paramterIdx;
+    return CANNON_PARAMETERS[idx];
+}
```

Comment:
```suggestion
const MapdataCannonPoint::CannonParameter &MapdataCannonPoint::parameters() const {
    assert(static_cast<size_t>(m_parameterIdx) < CANNON_PARAMETERS.size());
    return CANNON_PARAMETERS[m_parameterIdx];
}
```
See the `KartMove` comment - this is not a method of `MapdataCannonPoint` and should be moved out.